pkg/server/server.go

```diff
@@ -69,193 +71,194 @@ const resyncPeriod = 10 * time.Hour
 // systemShardCluster is the name of a logical cluster on every shard (including the root shard) that holds essential system resources (like the root APIs).
 var systemShardCluster = logicalcluster.New("system:shard")
 
-// Server manages the configuration and kcp api-server. It allows callers to easily use kcp
-// as a library rather than as a single binary. Using its constructor function, you can easily
-// setup a new api-server and start it:
-//
-//   srv := server.NewServer(server.BaseConfig())
-//   srv.Run(ctx)
-//
-// You may optionally provide PostStartHookFunc and PreShutdownHookFunc hooks before starting
-// the server that should be passed to the api-server itself. These hooks have access to a
-// restclient.Config which allows you to easily create a client.
-//
-//   srv.AddPostStartHook("my-hook", func(context genericapiserver.PostStartHookContext) error {
-//       client := clientset.NewForConfigOrDie(context.LoopbackClientConfig)
-//   })
-type Server struct {
-	options *kcpserveroptions.CompletedOptions
+type Config struct {
+	Options *kcpserveroptions.CompletedOptions
 
-	postStartHooks   []postStartHookEntry
-	preShutdownHooks []preShutdownHookEntry
+	EmbeddedEtcd *embeddedetcd.Config
 
-	syncedCh chan struct{}
+	GenericConfig  *genericapiserver.Config // the config embedded into MiniAggregator, the head of the delegation chain
+	MiniAggregator *aggregator.MiniAggregatorConfig
+	Apis           *apis.Config
+	ApiExtensions  *apiextensionsapiserver.Config
 
-	kcpSharedInformerFactory              kcpexternalversions.SharedInformerFactory
-	kubeSharedInformerFactory             coreexternalversions.SharedInformerFactory
-	apiextensionsSharedInformerFactory    apiextensionsexternalversions.SharedInformerFactory
-	dynamicDiscoverySharedInformerFactory *informer.DynamicDiscoverySharedInformerFactory
+	ExtraConfig
+}
 
-	// TODO(sttts): get rid of these. We have wildcard informers already.
-	rootKcpSharedInformerFactory  kcpexternalversions.SharedInformerFactory
-	rootKubeSharedInformerFactory coreexternalversions.SharedInformerFactory
+type ExtraConfig struct {
+	// resolveIdenties is to be called on server start until it succeeds. It injects the kcp
+	// resource identities into the rest.Config used by the client. Only after it succeeds,
+	// the clients can wildcard-list/watch most kcp resources.
+	resolveIdentities func(ctx context.Context) error // nolint:structcheck
+
+	// authentication
+	kcpAdminToken, shardAdminToken string // nolint:structcheck
```

Comment:
IMO just turn off the check at the linter level, so when they fix their bugs and we turn it back on we get the full checking?

Reply:
Not sure we want to chase their false positives across the codebase (is it even deterministic?)

Reply:
If I just bump the version?

Reply:
done